R/db-sql.R

```diff
@@ -281,6 +281,29 @@ sql_query_join.DBIConnection <- function(con, x, y, vars, type = "inner", by = N
     con = con
   )
 }
+#' @export
+sql_query_join.SQLiteConnection <- function(con, x, y, vars, type = "inner", by = NULL, na_matches = FALSE, ...) {
+  if (type == "full") {
+    # workaround as SQLite doesn't support FULL JOIN
+    build_sql(
+      sql_query_join(con, x, y, vars, type = "left", by = by, na_matches = na_matches, ...),
+      "UNION\n",
+      sql_query_join(con, y, x, vars, type = "left", by = by, na_matches = na_matches, ...),
+      con = con
+    )
+  } else if (type == "right") {
+    build_sql(
+      sql_query_join(
+        con, y, x, vars, type = "left", by = by, na_matches = na_matches,
```

Comment:
I have a vague recollection that just switching x and y messes up the variable names/order?

Reply:
in the generated query y is named RHS and x LHS which might be a bit confusing. But naming them the other way round messes up the select query. I added a comment about that in the code as one might be tempted to change the names of x and y.